File: src/main/java/com/example/eowa/model/Opinion.java
package com.example.eowa.model;

import jakarta.persistence.*;

@Entity
public class Opinion {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @ManyToOne
    private User user;

    @Column
    private UserOpinion userOpinion;

    public Opinion() {
    }

    public Opinion(User user, UserOpinion userOpinion) {
        this.user = user;
        this.userOpinion = userOpinion;
    }

    public long getId() {
        return id;
    }
    public User getUser() {
        return user;
    }

    public UserOpinion getUserOpinion() {
        return userOpinion;
    }

    public void setUserOpinion(UserOpinion userOpinion) {
        this.userOpinion = userOpinion;
    }

    public static enum UserOpinion{
        GOOD,
        BAD,
        TOLERABLE
    }
}

File: eowa_frontend/src/app/Model/Opinion.ts
import { User } from './User';

export interface Opinion {
  number?: number;
  id?: number;
  user: User;
  userOpinion: UserOpinion;
}

export enum UserOpinion {
  GOOD = 'GOOD',
  BAD = 'BAD',
  TOLERABLE = 'TOLERABLE',
}

export const UserOpinionLookup = {
  GOOD: UserOpinion.GOOD,
  BAD: UserOpinion.BAD,
  TOLERABLE: UserOpinion.TOLERABLE,
};
